File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0, profit = 0;
        int buyDay = prices[0];

        for(int i=1; i<prices.length; i++){
            if(prices[i] < buyDay){
                buyDay = prices[i];
            }
            else if(prices[i] > buyDay){
                profit = prices[i]-buyDay;
                maxProfit = Math.max(profit, maxProfit);
            }
        }

        return maxProfit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int arrLength = prices.size();
        int profit = 0, maxProfit = 0;

        int buy = prices[0];
        for (int index=1; index < arrLength; index++){
            if(prices[index] < buy){
                buy = prices[index];
            }else{
                profit = prices[index] - buy;
                maxProfit = max(profit, maxProfit);
            }
        }

        return maxProfit;
    }
};